src/manager/YaraManager.js

```diff
@@ -0,0 +1,124 @@
+import yara from 'yara';
+import { promisify } from 'util';
+
+export default class YaraManager {
+  constructor() {
+    this.resources = [];
+  }
+
+  /*
+   * This Method scans a list of raw js artifacts using
+   * a set of Yara rules. Returns the Js artifacts that match
+   * otherwise, no malware was detected.
+   *
+   * If an error occurs in processing, values are pushed to the RO
+   * in order to be handled correctly by our api.
+   *
+   * When pushing errors to the RO data buffer:
+   *      error: boolean (whether or not an error has occurred in processing)
+   *      matchedRule: only valid if (isMalicious == true)
+   *      isMalicious: boolean, null if unknown
+   */
+  async setupResourceScan(jsArtifacts, ticket) {
+    console.log('Scanning JS...');
+    const ticketId = ticket.getID();
+    const responseFile = 'yara_response.txt';
+    const initialize = promisify(yara.initialize);
+
+    try {
+      // link to yara binary
+      await initialize();
+
+      const rules = [{ filename: 'src/resources/rules.yara' }];
+      const scanner = yara.createScanner();
+      const warnings = await new Promise((resolve, reject) =>
+        scanner.configure({ rules }, (error, warnings) =>
+          error ? reject(error) : resolve(warnings)
+        )
+      );
+
+      if (warnings.length) {
+        console.log('Compile warnings: ' + JSON.stringify(warnings));
+
+        this.resources.push(
+          this.createYaraArtifact(ticketId, responseFile, true)
+        );
+      } else {
+        let matchFlag = false;
+
+        for (const js of jsArtifacts) {
+          const buf = { buffer: Buffer.from(js) };
```

Comment:
Hey Will this looks great so far. One thing is that our js here is a JSON object, we should specify the data field `js.data`. Also, since it is already a buffer, there shouldn't be a need to call Buffer.

Reply:
I think haha

Reply:
Changed in the latest copy, thanks for the heads-up!